File: python/bubble_sort.py
sequence = [4, 3, 5, 0, 1]

# Your Code Here
def bubble_sort(list_of_nums):
    swaps = 0
    is_swapped = True
    while is_swapped:
        is_swapped = False
        for idx in range(len(list_of_nums)-1):
            if list_of_nums[idx] > list_of_nums[idx+1]:
                swaps += 1
                is_swapped = True
                list_of_nums[idx], list_of_nums[idx + 1] = list_of_nums[idx+1], list_of_nums[idx]

    return list_of_nums, swaps


print(f"Final result: {bubble_sort(sequence)}")
# print(f"Swaps: {swaps}")

File: js/bubbleSort.js
var sequence = [4, 3, 5, 0, 1]
var swaps = 0

// Your Code Here
function result (numberList) {
    let isSwapped = true
    while (isSwapped) {
        isSwapped = false
        for (let i = 0; i < numberList.length; i++){
            let prev = numberList[i]
            let curr = numberList[i+1]
            if (numberList[i] > numberList[i+1]){
                numberList[i] = curr
                numberList[i+1] = prev
                isSwapped = true
                swaps ++
            }
        }
    }
    return numberList
}



console.log("Final result: " + result(sequence))
console.log("Swaps: " + swaps)
